### DISCUSSION TITLE
how do you play your games with the current adapter?

### DISCUSSION THREAD

**Discussion by opensourcerebel** [necroware/gameport-adapter#80] [General]
The current adapter works pretty good (all my M$ joysticks and FFB wheels work) I had some relative success with using it under ubuntu 18.04 to play Starwars Racer/ColinMcRae rally or Flatout 2 (ubuntu 20.04 and 22.04 was complete trash). On 18.04 it works decent but the gravity in SWR is a bit messed up and sometimes you stick to the walls. Flatout2 sometimes becomes choppy.
I decided to install win7 which worked on of my old machines ... but then I was immediately in driver hell :)

So I have decided to spare my self the exploration and ask around ... so how do you play these old games to get a better experience from your controllers (old pc + all the drivers or some linux with wine or?)?

**Comment by necroware** [comment]
I play on Linux. Use my joysticks for native modern games or wine for older windows games and even in DosBox for Dos games. But I also use Linux since the end of 90's, so issues are very rare here. I have to admit, that some games do work fine with joysticks and some are not well optimized for that. So, it depends on a game. I tested this GamePort adapter against native USB joysticks and there is no difference. If a game works good with a joystick, then it works also fine with the adapter. As for StarWars Racer, I love to play that game with a joystick, it makes so much more fun, than with a keyboard and/or mouse. I even once recorded a video playing that game with my Logitech Wingman Extreme: 

https://youtu.be/947DewHwbsE?list=PLif6qKQRcpVapZZSACNinfLxv4r7Ig0tG&t=631

**Comment by creopard** [comment]
I'm playing on Windows 11. Since Microsoft dropped the Gameport support from Windows Vista onwards, this adapter is the only real possibility to reuse all axes/buttons/hats on a modern Windows system.
There are a few modern games that insist on using a PS-like modern GamePad where classic (and even USB) joysticks are not recognised by the game itself.
But this behavior is not the fault of the OS, the adapter or the joystick but of the game that desperately needs dozens of buttons and axes.

**Comment by opensourcerebel** [comment]
@necroware Can you share your Linux setup? I use linux since the 2000s :) 
Getting wine to work on ubuntu 20 and 22 was a PITA ... but I am more of a ubuntu/debian person - what is your setup?

p.s.
I have a FFB which is native usb and one which is the digitar protocol which I managed to revive with your adapter. I can also share that there is no difference I can feel.

**Comment by opensourcerebel** [comment]
@creopard did you managed to run old classing on win 11? Like SWR or Flatout or ColinMcRae?

**Comment by necroware** [comment]
I don't think that it depends so much on the distribution if the one you use is up to date enough. My personal system is ArchLinux and I use the same setup since 2007 with xfce. The hardware is a Ryzen 5, AMD RX6600. Latest updates once a week, not much to say. At work I use mostly Ubuntu which I don't like for various reasons, but sometimes it is possible to use Debian-Testing there instead. That one would be my second best choice if ArchLinux would not exist, because I prefer rolling release strategy. Still, I don't think, that this has so much weight and I don't support distribution hopping. Everybody should understand the main differences once and stick to the favorite distribution. Every one of them have pro and contra, but most of the problems are not to resolve with re-installation anyway and should be understood and solved in depth. Once it's done, everything just works with any distribution.

In regards of joysticks, well, as I said, some games were made with joysticks in mind and some not. And I think, that it doesn't even make a difference if you are on Windows or Linux. Sometimes I have to setup a windows PC for my YT channel for testing and usually I use Snappy Driver Installer to automatically install everything in one go. It's worth it to try if you want to avoid the work: https://sdi-tool.org/

**Comment by creopard** [comment]
@opensourcerebel unfortunately, I'm not much into racing games, but if you configure the adapter (dip switches) accordingly, you can see, calibrate and test the joystick with the control panel program "joy.cpl". This program exists from Windows 95 up to Windows 11. (You will have to search for it in Windows 10/11 as it's kind of hidden and not linked directly in the control panel or the start menu.)

You can even use a classic joystick/gamepad to play with it on an Android phone:
https://www.youtube.com/watch?v=bDuXLQjaocE
